Rename variables and add comments

pietern opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #27531 Run clang-format for torch/distributed/rpc
* #27530 Rename PythonUDF{Call,Resp}
* #27529 Scope pybind11 functions to torch.distributed.{autograd,rpc}
* #27290 Move RPC API to torch.distributed.rpc
* #27289 Move internal functions to torch.distributed.rpc
* #27288 Move RPC backend registry to torch.distributed.rpc
* #27287 Remove torch.distributed.rpc function
* **#27286 Rename variables and add comments**
* #27285 Remove shebang from non-executable files in torch.distributed
* #27284 Fix pybind11 warnings in python_rpc_handler.cpp

The name `runUDFFunction` stutters because the F in UDF also stands
for function. Renamed these variables to be identical to their Python
equivalents. Renamed those to share a prefix and drop `internal`,
because internal functions can use an underscore prefix.

Differential Revision: [D17808208](https://our.internmc.facebook.com/intern/diff/D17808208)

pietern committed:
Rename variables and add comments

The name `runUDFFunction` stutters because the F in UDF also stands
for function. Renamed these variables to be identical to their Python
equivalents. Renamed those to share a prefix and drop `internal`,
because internal functions can use an underscore prefix.

xush6528 commented on torch/csrc/distributed/rpc/python_rpc_handler.cpp:
```diff
@@ -22,8 +22,8 @@ PythonRpcHandler::PythonRpcHandler() {
       py::module::import("torch.distributed.internal_rpc_utils");
-  runUDFFunction_ = getFunction(module, "run_python_udf_internal");
-  loadResultFunction_ = getFunction(module, "load_python_udf_result_internal");
+  pythonUDFRun_ = getFunction(module, "_python_udf_run");
```
Do you think "Python" + "UDF" is a good name here?

It's just any Python function.


I learned the word UDF in SQL, SparkSQL, where UDF is somoething beyond the built-in capability of the domain-specific language.

pietern replied:
Good point. I'll use @mrshenli's suggestion below.

xush6528 commented on torch/distributed/internal_rpc_utils.py:
```diff
@@ -99,11 +99,13 @@ def deserialize(self, binary_data, tensor_table):
+    Unpickles a Python function and its arguments, runs the function,
+    and pickles and returns the function's return value.
     """
     python_udf = _internal_rpc_pickler.deserialize(pickled_python_udf, tensors)
```
The name "_python_udf_run" doesn't indicate it's actually taking a serialized python function with args, deserialize it, run it, and serialize it.

Should we call it something else?

zhaojuanmao replied:
_python_udf_remote_run?

mrshenli replied:
Let's try to avoid using `remote` here, as `dist.remote` is another API with a different scope. 

In the message level, we have `PYTHON_CALL` and `PYTHON_RET`. Shall we use a similar name here? For example:

* `_python_udf_run` -> `_run_serialized_python_call`
* `load_python_udf_result_internal ` -> `_deserialize_python_ret`

pietern replied:
Good points. I stuck with the original naming because I didn't know about the intent here. As it can be any Python function, I think @mrshenli's suggestions are good.

zhaojuanmao reviewed (approved):
thanks for improving it, lgtm

pietern committed:
Update on "Rename variables and add comments"


The name `runUDFFunction` stutters because the F in UDF also stands
for function. Renamed these variables to be identical to their Python
equivalents. Renamed those to share a prefix and drop `internal`,
because internal functions can use an underscore prefix.

pietern commented:
@mrshenli @zhaojuanmao @xush6528 Renamed to remove UDF and added `py` prefix to indicate the C++ code is calling into Python code (which is good to know).

zhaojuanmao reviewed (approved):
lgtm

xush6528 reviewed (approved):
Thanks. Looks better!
